biggerquery/deploy.py

```diff
@@ -0,0 +1,111 @@
+    print('')
+
+
+def deploy_docker_image(build_ver: str, deployment_config, auth_method: str = 'local_account'):
+    docker_repository = deployment_config.resolve()['docker_repository']
+    docker_image = docker_repository + ":" + build_ver
+
+    print(f"Deploying docker image tag={docker_image} auth_method={auth_method}")
+
+    if auth_method == 'local_account':
+        os_call(['gcloud', 'auth', 'configure-docker'])
+    elif auth_method == 'service_account':
+        oauthtoken = get_vault_token(deployment_config)
+        os_call(['docker', 'login', '-u', 'oauth2accesstoken', '--password-stdin', 'https://eu.gcr.io'],
+                input=oauthtoken)
+    else:
+        raise ValueError('unsupported auth_method: ' + auth_method)
+
+    os_call(['docker', 'push', docker_image])
+    return docker_image
+
+
+def deploy_dags_folder(workdir: str, deployment_config: Config, clear_dags_folder=False,
+                       auth_method: str = 'local_account', gs_client=None):
+
+    print(f"Deploying DAGs folder, auth_method={auth_method}, clear_dags_folder={clear_dags_folder}, workdir={workdir}")
+    dags_bucket = deployment_config.resolve()['dags_bucket']
+
+    client = gs_client or create_storage_client(auth_method, deployment_config)
+    bucket = client.bucket(dags_bucket)
+
+    if clear_dags_folder:
+        clear_remote_DAGs_bucket(bucket)
+
+    upload_DAGs_folder(workdir, bucket)
+    return dags_bucket
+
+
+def clear_remote_DAGs_bucket(bucket):
+    i = 0
+    for blob in bucket.list_blobs(prefix='dags'):
+        if not blob.name in ['dags/', 'dags/airflow_monitoring.py']:
+            print(f"deleting file {blob_URI(blob)}")
+            blob.delete()
+            i += 1
+
+    print(f"{i} files deleted")
+
+
+def blob_URI(blob):
+    return f"gs://{blob.bucket.name}/{blob.name}"
+
+
+def upload_DAGs_folder(workdir: str, bucket):
+    dags_dir_path = get_dags_output_dir(workdir)
+
+    def upload_file(local_file_path, target_file_name):
+        print('bolb', target_file_name)
+        blob = bucket.blob(target_file_name)
+        blob.upload_from_filename(local_file_path, content_type='application/octet-stream')
+        print(f"uploading file {local_file_path} to {blob_URI(blob)}")
+
+    i = 0
+    for f in dags_dir_path.iterdir():
+        if f.is_file():
+            upload_file(f.as_posix(), 'dags/' + f.name)
+            i += 1
+
+    print(f"{i} files uploaded")
+
+
+def create_storage_client(auth_method: str, deployment_config):
+    project_id = deployment_config.resolve()['deploy_project_id']
+    if auth_method == 'local_account':
+        return storage.Client(project=project_id)
+    elif auth_method == 'service_account':
+        oauthtoken = get_vault_token(deployment_config)
+        return storage.Client(project=project_id, credentials=credentials.Credentials(oauthtoken))
+    else:
+        raise ValueError('unsupported auth_method: ' + auth_method)
+
+
+def get_vault_token(deployment_config):
```

Comment:
Do we have to use the Config class as a value object for these functions? 

Remember that Config won't be available during deployment (in a deployment plan the source is not available, parameters will be passed through the CLI). 

The second thing is that it's hard to say what the function needs (you need to read the function code). 

Thirdly, if we change something in the Config API, it will break the deployment.